app/views/devise/shared/_links.html.erb

```diff
@@ -3,16 +3,24 @@
 <% end %>
 
 <%- if devise_mapping.omniauthable? %>
-  <%- resource_class.omniauth_providers.each do |provider| %>
+ 
     <%= form_for "Login with Google",
-        url: omniauth_authorize_path(resource_name, provider), 
+        url: omniauth_authorize_path(resource_name, resource_class.omniauth_providers[0]), 
```

Comment:
This depends on the order of the array. Can't you call it directly by name?

Reply:
Good catch